### PR TITLE
pygame.transform.invert

### PR DESCRIPTION
**Pull request by ScriptLineStudios**
as per @novialriptide issue opened here: https://github.com/pygame/pygame/issues/3578
```python
pygame.transform.invert(surf: pygame.Surface, dest_surf: None) -> pygame.Surface
```

### CODE REVIEW

File: src_c/transform.c [hunk 1]
```diff
@@ -2966,6 +2966,82 @@ surf_average_color(PyObject *self, PyObject *args, PyObject *kwargs)
...
+    if (!newsurf) {
+        return NULL;
+    }
+
+    if (surfobj2) {
+        Py_INCREF(surfobj2);
+        return (PyObject *)surfobj2;
+    }
+    else {
+        return (PyObject *)pgSurface_New(newsurf);
```
Comment (novialriptide) [inline src_c/transform.c:3041]: Code quality nitpick: Could you remove the else statement and just move the return at the bottom of the function?

**Comment by MyreMylar** [comment]
Rebuilding on the CI to clear out python 3.6 test runs.

File: src_c/doc/transform_doc.h [hunk 2]
```diff
@@ -14,6 +14,7 @@
 #define DOC_PYGAMETRANSFORMLAPLACIAN "laplacian(surface, dest_surface=None) -> Surface\nfind edges in a surface"
 #define DOC_PYGAMETRANSFORMAVERAGESURFACES "average_surfaces(surfaces, dest_surface=None, palette_colors=1) -> Surface\nfind the average surface from many surfaces."
 #define DOC_PYGAMETRANSFORMAVERAGECOLOR "average_color(surface, rect=None, consider_alpha=False) -> Color\nfinds the average color of a surface"
+#define DOC_PYGAMETRANSFORMINVERT "average_color(surface, dest_surface=None) -> Surface\ninverts the RGB elements of a surface"
```
Comment (MyreMylar) [inline src_c/doc/transform_doc.h:17]:
```suggestion
#define DOC_PYGAMETRANSFORMINVERT "invert(surface, dest_surface=None) -> Surface\ninverts the RGB elements of a surface"
```

Docs are slightly wrong here.

**Review by MyreMylar — APPROVED** [review]
Test program used:

```
import pygame

pygame.init()


display_surf = pygame.display.set_mode((800, 600))

walt = pygame.image.load("images/walt.png").convert_alpha()
walt_invert = pygame.transform.invert(walt)

running = True

while running:

    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            running = False

    display_surf.fill((255, 255, 255))
    display_surf.blit(walt, (0, 0))
    display_surf.blit(walt_invert, (400, 0))

    pygame.display.flip()

```

Output:

image

Small error in the docs that should be fixed, but otherwise LGTM 👍

File: src_c/doc/transform_doc.h [hunk 3]
```diff
@@ -80,6 +81,10 @@ pygame.transform.average_color
  average_color(surface, rect=None, consider_alpha=False) -> Color
 finds the average color of a surface
 
+pygame.transform.invert
+ average_color(surface, dest_surface=None) -> Surface
```
Comment (MyreMylar) [inline src_c/doc/transform_doc.h:85]:
```suggestion
 invert(surface, dest_surface=None) -> Surface
```

**Comment by novialriptide** [comment]
Looks like you just need to format this pull request then we're all done here.

**Review by oddbookworm — APPROVED** [review]
I added a small commit to fix the doc header that the lint check was complaining about, so LGTM

**Pull request merged**